core/math/transform_2d.h

```diff
@@ -103,7 +103,7 @@ struct _NO_DISCARD_ Transform2D {
 	Transform2D looking_at(const Vector2 &p_target) const;
 
 	bool operator==(const Transform2D &p_transform) const;
-	bool operator!=(const Transform2D &p_transform) const;
+	INEQUALITY_OPERATOR(const Transform2D &)
```

Comment:
Doesn't this change the functionality in c++17?

Previously, the columns of the Transform2D were compared.
With this change, only the pointers to the two Transform2D are compared.

Lets take for example two different Transform2D `t1` and `t2`, which contain the same transform.

- previously `t1 != t2` was `false`
- with these changes `t1 != t2` is `true`.

Or to put it differently:
- with these changes `t1 == t2` is `true`
- with these changes `t1 != t2` is `true`.

Reply:
The inequality operator is just:
```cpp
_ALWAYS_INLINE_ bool operator!=(m_type p_other) const { return !(*this == p_other); }
```

Reply:
Thanks, looks like a misunderstanding on my side.